persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/ClinicalDataMapper.xml

```diff
@@ -53,12 +53,24 @@
                 cancer_study.CANCER_STUDY_IDENTIFIER = #{studyIds[0]}
             </if>
             <if test="sampleIds != null">
-                <foreach index="i" collection="sampleIds" open="(" separator="OR" close=")">
-                    cancer_study.CANCER_STUDY_IDENTIFIER = #{studyIds[${i}]} AND sample.STABLE_ID = #{sampleIds[${i}]}
-                </foreach>
+                <if test="studyIds.size() == 1">
+                    cancer_study.CANCER_STUDY_IDENTIFIER = #{studyIds[0]} AND
+                    sample.STABLE_ID IN
+                    <foreach item="item" collection="sampleIds" open="(" separator="," close=")">
+                        #{item}
+                    </foreach>
+                </if>
+                <if test="studyIds.size() > 1">
+                    <foreach index="i" collection="sampleIds" open="(" separator="OR" close=")">
```

Comment:
You can also use `IN` for pairs:

https://dba.stackexchange.com/questions/34266/selecting-where-two-columns-are-in-a-set